north/src/linux/mount.rs

```diff
@@ -12,152 +12,42 @@
-
-        /// When a file on this file system is accessed, only update the file's last access
-        /// time (atime) if the current value of atime is less than or equal to the file's
-        /// last modification time (mtime) or last status change time (ctime). This option is
-        /// useful for programs, such as mutt(1), that need to know when a file has been read
-        /// since it was last modified. Since Linux 2.6.30, the kernel defaults to the behavior
-        /// provided by this flag (unless MS_NOATIME was specified), and the MS_STRICTATIME
-        /// flag is required to obtain traditional semantics. In addition, since Linux 2.6.30,
-        /// the file's last access time is always updated if it is more than 1 day old.
-        const RELATIME = MS_RELATIME;
-
-        /// Remount an existing mount. This allows you to change the mountflags and data of an
-        /// existing mount without having to unmount and remount the file system. target should
-        /// be the same value specified in the initial mount() call; source and filesystemtype
-        /// are ignored.
-        ///
-        /// The following mountflags can be changed: MS_RDONLY, MS_SYNCHRONOUS, MS_MANDLOCK;
-        /// before kernel 2.6.16, the following could also be changed: MS_NOATIME and
-        /// MS_NODIRATIME; and, additionally, before kernel 2.4.10, the following could also
-        /// be changed: MS_NOSUID, MS_NODEV, MS_NOEXEC.
-        const REMOUNT = MS_REMOUNT;
-
-        /// Make this mount point shared.  Mount and unmount events immediately under this mount
-        /// point will propagate to the other mount points that are members of this mount's peer
-        /// group. Propagation here means that the same mount or unmount will automatically occur
-        /// under all of the other mount points in the peer group.  Conversely, mount and unmount
-        /// events that take place under peer mount points will propagate to this mount point.
-        const SHARED = MS_SHARED;
-
-        /// Suppress the display of certain (printk()) warning messages in the kernel log.
-        /// This flag supersedes the misnamed and obsolete MS_VERBOSE flag (available
-        /// since Linux 2.4.12), which has the same meaning.
-        const SILENT = MS_SILENT;
-
-        /// Always update the last access time (atime) when files on this file system are
-        /// accessed. (This was the default behavior before Linux 2.6.30.) Specifying this
-        /// flag overrides the effect of setting the MS_NOATIME and MS_RELATIME flags.
-        const STRICTATIME = MS_STRICTATIME;
-
-        /// Make writes on this file system synchronous (as though the O_SYNC flag to
-        /// open(2) was specified for all file opens to this file system).
-        const SYNCHRONOUS = MS_SYNCHRONOUS;
-    }
-}
+use std::ffi::CString;
 
 pub async fn mount(
     source: &Path,
     target: &Path,
     fstype: &str,
-    flags: MountFlags,
+    flags: nix::mount::MsFlags,
```

Comment:
Hm. Using foreign types in public api feels a little bit odd. 
Reexport them with 
```
pub use nix::mount::MsFags as Flags;
```